tests/drivers/spi/spi_loopback/src/spi.c

```diff
@@ -621,7 +621,11 @@ static void spi_loopback_test_word_size(struct spi_dt_spec *spec,
 	struct spi_config config_copy = spec->config;
 
 	config_copy.operation &= ~SPI_WORD_SIZE_MASK;
-	config_copy.operation |= SPI_WORD_SET(word_size);
+	if (word_size == 16) {
+		config_copy.operation |= SPI_WORD_SET(word_size - 1);
+	} else {
+		config_copy.operation |= SPI_WORD_SET(word_size);
+	}
```

Comment:
this is not correct, there shouldn't be something special in the API about 16 bits, we want to test it, not 15 bits. If your implementation is not passing, it's not a fault of the test